pystac/collection.py

```diff
@@ -97,7 +99,8 @@ def clone(self) -> SpatialExtent:
         Returns:
             SpatialExtent: The clone of this object.
         """
-        return SpatialExtent(
+        cls = self.__class__
+        return cls(
```

Comment:
If subclassed, I see how `cls = self.__class__` is the subclass. But are we concerned with the return being typed as the parent class (`SpatialExtent`) rather than the subclass? I'm trying to reconcile this with [#862](https://github.com/stac-utils/pystac/issues/862).